contracts/PoolManager.sol

```diff
@@ -331,11 +347,22 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
 
     /// @inheritdoc IPoolManager
     function settle(Currency currency) external payable override noDelegateCall onlyByLocker returns (uint256 paid) {
-        uint256 reservesBefore = reservesOf[currency];
-        reservesOf[currency] = currency.balanceOfSelf();
-        paid = reservesOf[currency] - reservesBefore;
+        Currency accountCurrency = (currency == WRAPPED_NATIVE) ? CurrencyLibrary.NATIVE : currency;
+        uint256 balance = currency.balanceOfSelf();
+
+        if (currency == WRAPPED_NATIVE) {
```

Comment:
same with moving into CurrencyLibrary

Mostly concerned with adding so much non-generic logic into these very simple generic functions

Reply:
also thinking now that we have custom accounting, would we be better off just having a CA ETH/WETH pool? and enforcing WETH pairs only against ETH

Reply:
Would we even need CA to do this?

Reply:
I was thinking a hook that noops swap and instead just takes ETH and wraps it into WETH.. could avoid CA by just adding a wrapETH command but this feels nicer since it fits nicely into swap paths and  were adding CA anywas

Reply:
I see.. but then dont you also have to enforce that the eth/weth pool is a CA pool if you want to avoid having weth inside the contract. And then Im not sure how you enforce that